internal/host/statemachine.go

```diff
@@ -34,6 +35,17 @@ func NewHostStateMachine(sm stateswitch.StateMachine, th *transitionHandler) sta
 		PostTransition:   th.PostRegisterHost,
 	})
 
+	// Register host by late binding
```

Comment:
You don't need to add this transaction.You can simply add models.HostStatusUnbinding state as on  of the source states to the RegisterHostTransaction in pool_state_machine.go file. After register the host state should be set to HostStatusDiscoveringUnbound.

Reply:
done